06_puppy/sirigithub/mapstore.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+)
+
+type MapStore map[int]Puppy
+
+func NewMapStore() MapStore {
+	return MapStore{}
+}
+
+// CreatePuppy takes a Puppy with a user provided ID, and creates a new Puppy if
+// the ID does not already exist in the store.
+func (m MapStore) CreatePuppy(puppy *Puppy) error {
+	if _, exists := m[puppy.ID]; exists {
```

Comment:
s/exists/ok ([comma ok idiom](https://golang.org/doc/effective_go.html#maps)), below as well.

Reply:
Typically IDs are provided by system (MapStore) not by client/user input - typically user has no knowledge of all/available IDs. typically IDs (or whole created puppy) is returned by create function.

Reply:
@juliaogris good point indeed. I have reworked my earlier lazy implementation ;)
The Map stores handle the ID creation and return the created ID.
I remove the id check logic from create methods as a new id is created for each insert.